### PR TITLE
Adapt Redis config template due to changes made to Redis config after service restart

### PR DESCRIPTION
**Pull request by christianhueserhzdr**
* With Redis v6.2.7 the Redis config file changed after restart of Redis service.
* Quoting scheme changed in redis.conf.

### CODE REVIEW

**Comment by christianhueserhzdr** [comment]
Dear Norman @Normo, I spotted that the Redis.conf changed after upgrading Redis to v6.2.7 and restarting the Redis service. Do you think this change is alright given the differences listed in issue #19? Thanks for reviewing this pull request.

**Comment by tobiashuste** [comment]
@christianhueserhzdr Do you have an idea how we can stay compatible with versions previous of `6.2.7`? For those, we would then ship a different template.

**Comment by christianhueserhzdr** [comment]
> @christianhueserhzdr Do you have an idea how we can stay compatible with versions previous of `6.2.7`? For those, we would then ship a different template.

Dear Tobias @tobiashuste, would it be enough to discriminate between the Redis version string and then ship different Redis config template files?

Something along these lines?

```yaml
- name: Check if version string is greater equal version {{ redis_version_config_change }}.
  ansible.builtin.set_fact:
    is_redis_version_newer: "{{ redis_version is version(redis_version_config_change, operator='>=', strict=True) }}"

- name: Provide newer Redis config template file if Redis version is greater equal {{ redis_version_config_change }}.
  when: is_redis_version_newer
  block:
    - name: Configure Redis for newer version of Redis.
      become: yes
      ansible.builtin.template:
        src: redis.conf.new.j2
        dest: "{{ redis_server_configuration_file }}"
        mode: 0640
        owner: "{{ redis_user }}"
        group: "{{ redis_group }}"
      notify: Restart Redis Server

- name: Provide older Redis config template file if Redis version is not greater equal {{ redis_version_config_change }}.
  when: not is_redis_version_newer
  block:
    - name: Configure Redis for older version of Redis.
      become: yes
      ansible.builtin.template:
        src: redis.conf.old.j2
        dest: "{{ redis_server_configuration_file }}"
        mode: 0640
        owner: "{{ redis_user }}"
        group: "{{ redis_group }}"
      notify: Restart Redis Server
```

**Comment by christianhueserhzdr** [comment]
Dear Tobias @tobiashuste, 

I refactored the snippet above a bit to remove redundancies. What do you think?

```yaml
- name: Check if version string is greater equal version {{ redis_version_config_change }}.
  ansible.builtin.set_fact:
    is_redis_version_newer: "{{ redis_version is version(redis_version_config_change, operator='>=', strict=True) }}"
    redis_conf_template_discriminator: "newer"

- name: Select older version of the Redis config template file.
  ansible.builtin.set_fact:
    redis_conf_template_discriminator: "older"
  when: not is_redis_version_newer

- name: Configure Redis depending on Redis version.
  become: yes
  ansible.builtin.template:
    src: "redis.conf.{{ redis_conf_template_discriminator }}.j2"
    dest: "{{ redis_server_configuration_file }}"
    mode: 0640
    owner: "{{ redis_user }}"
    group: "{{ redis_group }}"
  notify: Restart Redis Serveryaml
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,9 @@ Group your changes into these categories:
 
 ### Changed
 
+- Adapt Redis config template due to changes made to Redis config after service restart
```
Comment (Normo) [inline CHANGELOG.md:25]: This is not needed, since we can use the github-changelog-generator.

File: templates/redis.conf.j2 [hunk 2]
```diff
@@ -7,8 +7,8 @@ supervised no
 daemonize yes
 pidfile "/var/run/redis/redis-server.pid"
 logfile "{{ redis_server_log_file_path }}"
-loglevel "{{ redis_log_level }}"
-dir /var/lib/redis/
+loglevel {{ redis_log_level }}
```
Comment (Normo) [inline templates/redis.conf.j2:10]: @christianhueserhzdr I was not able to reproduce this issue when upgrading from `6.2.6` to `6.2.7`. 

I tried the following steps:
1. `git checkout main`
2. Ensure `redis_version` in `defaults/main.yml` is `6.2.6`
3. `pipenv run molecule converge`
4. Optional: `pipenv run molecule login` and view the redis-server config
5. Change `redis_version` in `defaults/main.yml` to `6.2.7`
6. `pipenv run molecule converge`
7. Check if config has changed

**Review by Normo — CHANGES_REQUESTED** [review]

File: templates/redis.conf.j2 (hunk 2 above)
Comment (tobiashuste) [reply templates/redis.conf.j2:10]: Do we need this change then, @christianhueserhzdr?

Comment (christianhueserhzdr) [reply templates/redis.conf.j2:10]: No, we don't need this PR any longer. Thanks, Tobias @tobiashuste, for the reminder. I'll close it now.

**Comment by christianhueserhzdr** [comment]
This PR has already been resolved. I'll close it now.

**Pull request closed without merging**